### PR TITLE
Add event logs for disposing local video stream

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
Add event trace logs for disposing local video stream

# Why
We have them for remote video streams and for creating local video streams, but not for disposing local video streams.

# How Tested
I did not as this is just adding logs - please review thoroughly

### CODE REVIEW

**Review by carocao-msft — APPROVED** [review]

File: packages/calling-stateful-client/src/StreamUtils.ts [hunk 1]
```diff
@@ -446,24 +446,58 @@ function disposeViewRemoteVideo(
...
     return;
   }
 
   // Sets the status and also renderer. I think we need to always set renderer to undefined since in all status when
   // cleaned up should have renderer as undefined. If the status is 'Rendered' and renderer is not defined it should
   // be cleaned up below so we can set it to undefined here.
   if (renderInfo.status === 'Rendering') {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_STREAM_STOPPING,
+      level: 'warning',
```
Comment (prprabhu-ms) [inline packages/calling-stateful-client/src/StreamUtils.ts:477]: Why is this a warning?

Stopping a stream that is in the process of being started is normal, right? It just means that the user clicked the video toggle button too soon.

File: packages/calling-stateful-client/src/StreamUtils.ts [hunk 2]
```diff
@@ -446,24 +446,58 @@ function disposeViewRemoteVideo(
...
+      name: EventNames.DISPOSING_LOCAL_RENDERER,
+      level: 'info',
+      message: 'Disposing local view renderer.',
+      data: streamLogInfo
+    });
     renderInfo.renderer.dispose();
+  } else {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_RENDERER_NOT_FOUND,
+      level: 'warning',
```
Comment (prprabhu-ms) [inline packages/calling-stateful-client/src/StreamUtils.ts:497]: Shouldn't this also be an `error`, similar to line 464?

File: packages/calling-stateful-client/src/StreamUtils.ts [hunk 3]
```diff
@@ -472,21 +506,48 @@ function disposeViewUnparentedVideo(
...
+      level: 'error',
+      message: 'Cannot find render info when disposing unparented local stream.'
+    });
     return;
   }
 
   if (renderInfo.status === 'Rendering') {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_STREAM_STOPPING,
+      level: 'warning',
```
Comment (prprabhu-ms) [inline packages/calling-stateful-client/src/StreamUtils.ts:530]: Same comment as above.

File: packages/calling-stateful-client/src/StreamUtils.ts [hunk 4]
```diff
@@ -472,21 +506,48 @@ function disposeViewUnparentedVideo(
...
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.DISPOSING_LOCAL_RENDERER,
+      level: 'info',
+      message: 'Disposing unparented local view renderer.'
+    });
     renderInfo.renderer.dispose();
+  } else {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_RENDERER_NOT_FOUND,
+      level: 'warning',
```
Comment (prprabhu-ms) [inline packages/calling-stateful-client/src/StreamUtils.ts:548]: Same comment as above.

It's unfortunate we have duplicate code-paths for such closely related non-trivial operations. It's an invitation to bugs...

File: packages/calling-stateful-client/src/StreamUtils.ts [hunk 5]
```diff
@@ -511,7 +572,11 @@ export function createView(
     // Render LocalVideoStream that is not part of a Call
     return createViewUnparentedVideo(context, internalContext, stream, options);
   } else {
-    console.warn('Invalid combination of parameters');
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.CREATE_STREAM_INVALID_PARAMS,
+      level: 'warning',
```
Comment (prprabhu-ms) [inline packages/calling-stateful-client/src/StreamUtils.ts:577]: I agree with this being a `warning`. Client used the API incorrectly, not an internal error.

**Review by prprabhu-ms — APPROVED** [review]

File: packages/calling-stateful-client/src/StreamUtils.ts (hunk 1 above)
Comment (JamesBurnside) [reply packages/calling-stateful-client/src/StreamUtils.ts:477]: Good note, updating here and remote rendering ops to be info

File: packages/calling-stateful-client/src/StreamUtils.ts (hunk 2 above)
Comment (JamesBurnside) [reply packages/calling-stateful-client/src/StreamUtils.ts:497]: Yep I agree, updating here and remote rendering ops to be error

File: packages/calling-stateful-client/src/StreamUtils.ts (hunk 3 above)
Comment (JamesBurnside) [reply packages/calling-stateful-client/src/StreamUtils.ts]: updated

File: packages/calling-stateful-client/src/StreamUtils.ts (hunk 4 above)
Comment (JamesBurnside) [reply packages/calling-stateful-client/src/StreamUtils.ts]: updated, yeah I agree on the redundancy but also can't commit time to making it better, the code is so similar we should be able to have a local | remote type and keep it somewhat agnostic

**Pull request merged**